src/main/java/com/mathworks/ci/MatlabBuild.java

```diff
@@ -29,28 +30,47 @@ public interface MatlabBuild {
      * @return matlabLauncher returns the process launcher to run MATLAB commands
      */
     default ProcStarter getProcessToRunMatlabCommand(FilePath workspace,
-            Launcher launcher, TaskListener listener, EnvVars envVars, String matlabCommand, String uniqueName)
+            Launcher launcher, TaskListener listener, EnvVars envVars, String matlabCommand, String startupOpts, String uniqueName)
             throws IOException, InterruptedException {
         // Get node specific temp .matlab directory to copy matlab runner script             
-        FilePath targetWorkspace = new FilePath(launcher.getChannel(),
-                workspace.getRemote() + "/" + MatlabBuilderConstants.TEMP_MATLAB_FOLDER_NAME);
+        FilePath targetWorkspace; 
         ProcStarter matlabLauncher;
         if (launcher.isUnix()) {
-            final String runnerScriptName = uniqueName + "/run_matlab_command.sh";
+            targetWorkspace = new FilePath(launcher.getChannel(),
+                workspace.getRemote() + "/" + MatlabBuilderConstants.TEMP_MATLAB_FOLDER_NAME);
+
+            // Determine whether we're on Mac on Linux
+            ByteArrayOutputStream kernelStream = new ByteArrayOutputStream();
+            launcher.launch()
+                .cmds("uname")
+                .masks(true)
+                .stdout(kernelStream)
+                .join();
+
+            String binaryName;
+            String runnerName = uniqueName + "/run-matlab-command";
+            if (kernelStream.toString("UTF-8").contains("Linux")) {

```

Comment:
This is probably fine, but I'm more used to special casing to mac and defaulting to Linux rather than visa versa. Not a huge deal though